app/models/page_view.rb

```diff
@@ -46,6 +46,14 @@ class PageView < ApplicationRecord
     customRanking ["desc(visited_at_timestamp)"]
   end
 
+  def self.trigger_delayed_job(record, remove)
+    if remove
+      record.delay.remove_from_index!
```

Comment:
Currently, we don't use DelayedJob `delay` and use ActiveJobs instead so that the possible migration to another background job queue would be easier. We refactored all the `delay` calls while working on the #3136, the whole issue is also described in #2497.
Another problem with this is that it will pass ActiveRecord object instead of an id, so the job could be executed when the record is not in the database already, so the job will fail with a deserialization error. I suppose `Search::RemoveFromIndexJob` and `Search::IndexJob` (added in #5063) can be used here.

Reply:
Ah, thank you for the insight! I figured you would know best for this PR 😄